### PR TITLE
Refs[TASK][34788] Handle click item list meal

### PR DESCRIPTION
**Pull request by GiangGiang1105**
## Related Tickets
- [#Task 34788: Handle click item list meal](https://edu-redmine.sun-asterisk.vn/issues/34788)
## WHAT
- Handle click item list meal
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/gfood/screen/listmeal/ListMealFragment.kt [hunk 1]
```diff
@@ -13,13 +13,16 @@ import com.example.gfood.data.source.repository.ListMealRepository
...
 import com.example.gfood.utils.loadImage
 import kotlinx.android.synthetic.main.fragment_list_meal.*
 
 class ListMealFragment : Fragment(), ListMealConstract.View {
 
     private val listMealAdapter by lazy {
-        ListMealAdapter()
+        ListMealAdapter() {
+            startActivity(MealDetailActivity.newIntent(context, it))
+        }
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/example/gfood/screen/listmeal/ListMealFragment.kt:25]: khởi tạo adapter mà tạo sao lại chuyển màn hình e?

File: app/src/main/java/com/example/gfood/screen/mealdetail/MealDetailActivity.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.example.gfood.R
+import com.example.gfood.data.model.Meal
+
+class MealDetailActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/example/gfood/screen/mealdetail/MealDetailActivity.kt]:
```suggestion
class MealDetailActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
```

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -11,6 +11,8 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.GFOOD">
+        <activity android:name=".screen.mealdetail.MealDetailActivity"
+            android:theme="@style/Theme.AppCompat.Light.NoActionBar"/>
         <activity
             android:name=".screen.main.MainActivity"
             android:theme="@style/Theme.AppCompat.Light.NoActionBar">
```
Comment (tuanlvt-1117) [inline app/src/main/AndroidManifest.xml]: 2 activity chung 1 theme.
Refactor code

File: app/src/main/java/com/example/gfood/screen/listmeal/ListMealFragment.kt (hunk 1 above)
Comment (GiangGiang1105) [reply app/src/main/java/com/example/gfood/screen/listmeal/ListMealFragment.kt:25]: Dạ vì em truyền higher function vào chỗ này để onclick á anh

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**